lib/src/client.dart

```diff
@@ -1600,12 +1608,24 @@ class Client extends MatrixApi {
         await firstSyncReceived;
       }
       return;
+    } on ClientInitException {
+      rethrow;
     } catch (e, s) {
-      Logs().e('Initialization failed', e, s);
-      await logout().catchError((_) => null);
+      Logs().wtf('Client initialization failed', e, s);
       onLoginStateChanged.addError(e, s);
+      final clientInitException = ClientInitException(
+        e,
+        homeserver: homeserver,
+        accessToken: accessToken,
+        userId: userID,
+        deviceId: deviceID,
+        deviceName: deviceName,
+        olmAccount: olmAccount,
+      );
+      await clear();
+      throw clientInitException;
+    } finally {
       _initLock = false;
```

Comment:
The init lock will never be released in this case.

Reply:
But I don't understand why. I thought so because it was not in the finally block before but I don't understand the reason because `finally{}` is supposed to be executed at any condition.

Like this in dartpad works:

```dart
void main() {
  try {
    print('Test');
    throw 'Bumm';
  } catch(e) {
    rethrow; // Same with `throw 'Bumm 2';`
  } finally {
    print('Finally trotzdem ausgeführt'); // Appears in the output
  }
}
```

Reply:
Hm, interesting. I guess that is my mistake then. It does NOT execute if there is a return, but it does execute when there is a throw... I have no idea why and no other language I know does that, so it always trips me up. I just don't think it is very intuitive to have control flow in a catch block AND a finally block. But this code is fine then, yes.